### PR TITLE
[`refurb`] Implement `redundant-log-base` (`FURB163`)

### PR DESCRIPTION
**Pull request by tjkuson**
## Summary

Implement [`simplify-math-log`](https://github.com/dosisod/refurb/blob/master/refurb/checks/math/simplify_log.py) as `redundant-log-base` (`FURB163`).

Auto-fixes

```python
import math

math.log(2, 2)
```

to

```python
import math

math.log2(2)
```

Related to #1348.

## Test Plan

`cargo test`

### CODE REVIEW

**Review by zanieb — APPROVED** [review]
Nice! This lgtm except the rule needs to go through preview.

File: crates/ruff_linter/src/codes.rs [hunk 1]
```diff
@@ -955,6 +955,7 @@ pub fn code_to_rule(linter: Linter, code: &str) -> Option<(RuleGroup, Rule)> {
         (Refurb, "145") => (RuleGroup::Preview, rules::refurb::rules::SliceCopy),
         (Refurb, "148") => (RuleGroup::Preview, rules::refurb::rules::UnnecessaryEnumerate),
         (Refurb, "152") => (RuleGroup::Preview, rules::refurb::rules::MathConstant),
+        (Refurb, "163") => (RuleGroup::Stable, rules::refurb::rules::RedundantLogBase),
```
[the remark this answers is not in the document]
Comment (tjkuson) [reply crates/ruff_linter/src/codes.rs:958]: Oops, must've missed that. Apologies!

**Comment by tjkuson** [comment]
Yeah, that's fair. I did think about which name to use when porting the rule, but never found one I fully liked (it has to both describe both the `math.log(foo, math.e)` case and the `math.log(foo, 2)` case. I settled for redundant log base to mean 'there is a way of writing this logarithmic function without specific a base' instead of 'just remove the base argument'. 

I also tried to make the rule message be explicit with what its recommendation is, to guard against confusion.

```
FURB163.py:15:1: FURB163 [*] Replace with `math.log2(1)`
```

Also, for what it's worth, the corresponding refurb rule [`simplify-math-log`](https://github.com/dosisod/refurb/blob/master/refurb/checks/math/simplify_log.py) cites readability, not performance, as the motivation. I cited both readability and precision when porting the rule in the documentation (as well as efficiency; I found the fixes ran faster).

**Comment by bluetech** [comment]
Got it on the name. So my only nit is: in the rule description, replace "precise" with "accurate"; both `math.log` and `math.log2` have the same FP precision (double), but the latter is more accurate (for base 2).

File: crates/ruff_linter/src/rules/refurb/rules/redundant_log_base.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+/// redundant.
+///
+/// Instead of passing 2 or 10 as the base, use `math.log2` or `math.log10`
+/// respectively. This is more readable, precise, and efficient.
+///
+/// ## Example
+/// ```python
+/// import math
+///
+/// math.log(2, 2)
```
[the remark this answers is not in the document]
Comment (tjkuson) [reply crates/ruff_linter/src/rules/refurb/rules/redundant_log_base.rs]: Changed to `3`.

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**